examples/05_scaling_non_linear_models.py

```diff
@@ -0,0 +1,537 @@
+# :code:`X_train` and :code:`y_train` get sliced into subsets of increasing
+# size, while :code:`X_test` and :code:`y_test` do not get changed when the
+# sample size varies.
+from sklearn.metrics import average_precision_score
+# define the different train set sizes on which to evaluate the model
+train_set_sizes = [train_set_size // 10, train_set_size // 3, train_set_size]
+
+def evaluate_model(model, X_train, y_train, X_test, y_test, train_set_sizes,
+                   one_hot_encoder):
+    train_times, test_scores = [], []
+
+    for n in train_set_sizes:
+        train_time, _ = profile(model.fit)(X_train[:n], y_train[:n])
+
+        y_pred = model.predict(X_test)
+
+        y_pred_onehot = one_hot_encoder.transform(y_pred.reshape(-1, 1))
+        y_test_onehot = one_hot_encoder.transform(y_test.reshape(-1, 1))
+
+        test_score = average_precision_score(y_test_onehot, y_pred_onehot)
+
+        train_times.append(train_time)
+        test_scores.append(test_score)
+    return train_times, test_scores
+
+train_times_svc, test_scores_svc = evaluate_model(model, X_train, y_train,
+                                                  X_test, y_test,
+                                                  train_set_sizes,
+                                                  one_hot_encoder)
+
+###############################################################################
+# Plotting the training accuracy and the time:
+import matplotlib.pyplot as plt
+def plot_time_and_scores(
+        train_set_sizes, train_times, test_scores, title=None):
+    f, ax_time = plt.subplots()
+    ax_time.set_ylabel('time')
+    ax_time.set_xlabel('training set size')
+    ax_time.plot(train_set_sizes, train_times, 'b-', label='time')
+    ax_time.legend(loc='upper left')
+
+    # display scores in the right-hand side y-axis
+    ax_score = ax_time.twinx()
+    ax_score.set_ylabel('score')
+    ax_score.plot(train_set_sizes, test_scores, 'r-', label='score')
+    ax_score.legend(loc='upper right')
+
+    if title is not None:
+        f.suptitle(title)
+    return f
+
+title = 'Test Accuracy and Fitting Time for a support vector classifier'
+f = plot_time_and_scores(train_set_sizes, train_times_svc, test_scores_svc,
+                         title=title)
+
+
+###############################################################################
+# Increasing training size cleary improves model accuracy. However, the
+# training time and the input size increase quadratically with the training set
+# size.  Indeed, kernel methods need to process an entire :math:`n \times n`
+# matrix at once. In order for this matrix to be loaded into memory, :math:`n`
+# has to remain low: Using 30000 observations, the input is a :math:`30000
+# \times 30000` matrix.  If composed of 32-bit floats, its total size is around
+# :math:`30000^2 \times 32 = 2.8 \times 10^{10} \text{bits} = 4\text{GB}`,
+# which is the caracteristic size of many personal laptops.
```

Comment:
I think that you can remove "which is the caracteristic size of many personal laptops". In general, it is useful to say less. It's a classic mantra of communication: "less is more", in other words, the less you say, the more what you say in impactful (I should learn that IRL :D ).